encodings/dict/src/compute.rs

```diff
@@ -49,7 +49,12 @@ impl TakeFn for DictArray {
 impl SliceFn for DictArray {
     // TODO(robert): Add function to trim the dictionary
     fn slice(&self, start: usize, stop: usize) -> VortexResult<Array> {
-        Self::try_new(slice(self.codes(), start, stop)?, self.values()).map(|a| a.into_array())
+        let new_codes = slice(self.codes(), start, stop)?;
+        if (stop - start) * 5 < self.values_len() {
+            take(self.values(), &new_codes)
```

Comment:
This will not produce a unique dictionary. You have to rekey the values you see in codes. There’s code in arrow c++ that does this the correct way

Reply:
I'm just take'ing the values directly rather than creating a new dict array.

I could instead trim the dictionary, but there must be a threshold at which it's faster to memcpy the repeated values than to de-duplicate the codes `new_codes` and `take` a new dictionary. I suppose we can't know if we reached the threshold until after we de-duplicate the codes.

Reply:
It’s hard to read this on a phone. I thought we are creating a new dict array. I agree on the general assumption but I would like to see a minimal benchmark

Reply:
Mostly we need to tune this constant so I would externalize it to some global const that’s in an obvious place

Reply:
Yeah, you're right. I should get data